src/org/pahappa/systems/exceptions/UpdateFailedException.java

```diff
@@ -0,0 +1,8 @@
+package org.pahappa.systems.exceptions;
+
+public class UpdateFailedException extends Exception{
+    UpdateFailedException(String message){
```

Comment:
I don't think we need an extra UpdateFailedException class

Reply:
on it